src/lib.rs

```diff
@@ -255,6 +237,40 @@ impl HidApi {
         }
     }
 
+    /// Open a HID device using libusb_wrap_sys_device.
+    pub fn wrap_sys_device(&self, sys_dev: isize, interface_num: i32) -> HidResult<HidDevice> {
+        if cfg!(not(any(
```

Comment:
Instead of throwing an error at runtime, just `cfg()` this method to make it available to libusb based backend only.

b.t.w. maybe also check if this libusb functionality is also available on other platforms?

Reply:
If this is available to libusb based backend only, library users who use this function also have to use cfg() in their codes.
So, I think it is better to just fail the function if other backends is used, isn't it?

Reply:
Libusb wrap_sys_device functionality is available on other platforms, but  the hidapi does not use libusb as a backend on other platforms.

Reply:
> Libusb wrap_sys_device functionality is available on other platforms, but the hidapi does not use libusb as a backend on win / macOS. So, freeBSD, openBSD can use this. However, build.rs setting for freeBSD/openBSD does not use features.

You can just use `cfg(target_os = "freebsd)", and similar for openbsd.

Reply:
> If this is available to libusb based backend only, library users who use this function also have to use cfg() in their codes. So, I think it is better to just fail the function if other backends is used, isn't it?

If you compile the function conditionally, the user of the library can use cfg(), and e.g. still opt into returning an error to the user at runtime. Also, the function is so specific to libusb, I do not really want to expose it to anyone else. I always prefer compile time errors over runtime.

Reply:
I see. I totaly misunderstood the feature system.